java/code/src/com/suse/manager/webui/controllers/utils/AbstractMinionBootstrapper.java

```diff
@@ -214,8 +223,62 @@ protected BootstrapResult bootstrapInternal(BootstrapParameters params, User use
         catch (Exception e) {
             return new BootstrapResult(false, Optional.empty(), e.getMessage());
         }
+        finally {
+            handleAnsibleCleanup(params, user);
+        }
+    }
+
+    /**
+     * Authenticate the Uyuni ssh public key on the bootstrapped host.
+     *
+     * @param params bootstrap params
+     * @param user the user
+     * @throws RuntimeException in case the action was not successful
+     */
+    private void handleAnsiblePreAuthentication(BootstrapParameters params, User user) {
+        LOG.info("Pre-authenticating system using Ansible inventory ID: " + params.getAnsibleInventoryId());
+        params.getAnsibleInventoryId()
+                .flatMap(pathId -> AnsibleManager.lookupAnsiblePathById(pathId, user))
+                .filter(path -> path instanceof InventoryPath)
+                .ifPresent(inventoryPath -> {
+                    Map<String, Object> pillar = Map.of(
+                            "user", params.getUser(),
+                            "inventory", inventoryPath.getPath().toString(),
+                            "target_host", params.getHost(),
+                            "ssh_pubkey", FileUtils.readStringFromFile(SaltSSHService.SSH_PUBKEY_PATH));
+
+                    LocalCall<Map<String, ApplyResult>> call =
+                            State.apply(List.of("ansible.mgr-ssh-pubkey-authorized"), of(pillar));
+                    String minionId = inventoryPath.getMinionServer().getMinionId();
+                    saltApi.callSync(call, minionId).ifPresentOrElse(
+                            res -> {
+                                // all results must be successful, otherwise we throw an exception
+                                List<String> failedStates = res.entrySet().stream()
+                                        .filter(r -> !r.getValue().isResult())
+                                        .map(r -> r.getKey())
+                                        .collect(Collectors.toList());
+
+                                if (!failedStates.isEmpty()) {
+                                    LOG.error("Ansible pre-authentication failed: " + failedStates);
```

Comment:
Could we add the `minionId` value to this `LOG.error` line? I believe it could be helpful understanding and investigating what went wrong. Or is this information present already in the `failedStates` content?